src/emulate.c

```diff
@@ -1240,27 +1244,6 @@ static block_t *block_find(const block_map_t *map, const uint32_t addr)
     return NULL;
 }
 
-/* execute a basic block */
-static bool block_emulate(riscv_t *rv, const block_t *block)
-{
-    const uint32_t n_insn = block->n_insn;
-    const rv_insn_t *ir = block->ir;
-
-    /* execute the block */
-    for (uint32_t i = 0; i < n_insn; i++) {
-        /* enforce zero register */
-        rv->X[rv_reg_zero] = 0;
```

Comment:
Register x0 should set zero before emulator executes every instruction. However, [riscv-arch-test](https://github.com/riscv-non-isa/riscv-arch-test) will be failed.